src/Umbraco.Infrastructure/Services/Implement/DistributedJobService.cs

```diff
@@ -93,45 +115,64 @@ public async Task FinishAsync(string jobName)
+        {
+            return;
+        }
+
         using ICoreScope scope = _coreScopeProvider.CreateCoreScope();
         scope.WriteLock(Constants.Locks.DistributedJobs);
 
         DistributedBackgroundJobModel[] existingJobs = _distributedJobRepository.GetAll().ToArray();
         var existingJobsByName = existingJobs.ToDictionary(x => x.Name);
 
-        foreach (IDistributedBackgroundJob registeredJob in _distributedBackgroundJobs)
+        // Collect all changes first, then execute - minimizes time spent in the critical section
+        var jobsToAdd = new List<DistributedBackgroundJobModel>();
+        DateTime utcNow = DateTime.UtcNow;
+
+        foreach (KeyValuePair<string, TimeSpan> registeredJob in registeredJobsByName)
         {
-            if (existingJobsByName.TryGetValue(registeredJob.Name, out DistributedBackgroundJobModel? existingJob))
+            if (existingJobsByName.TryGetValue(registeredJob.Key, out DistributedBackgroundJobModel? existingJob))
             {
-                // Update if period has changed
-                if (existingJob.Period != registeredJob.Period)
+                // Update only if period has actually changed
+                if (existingJob.Period != registeredJob.Value)
                 {
-                    existingJob.Period = registeredJob.Period;
+                    existingJob.Period = registeredJob.Value;
                     _distributedJobRepository.Update(existingJob);
                 }
             }
             else
             {
-                // Add new job (fresh install or newly registered job)
-                var newJob = new DistributedBackgroundJobModel
+                // Collect new jobs for batch insert
+                jobsToAdd.Add(new DistributedBackgroundJobModel
                 {
-                    Name = registeredJob.Name,
-                    Period = registeredJob.Period,
-                    LastRun = DateTime.UtcNow,
+                    Name = registeredJob.Key,
+                    Period = registeredJob.Value,
+                    LastRun = utcNow,
                     IsRunning = false,
-                    LastAttemptedRun = DateTime.UtcNow,
-                };
-                _distributedJobRepository.Add(newJob);
+                    LastAttemptedRun = utcNow,
+                });
             }
         }
 
-        // Remove jobs that are no longer registered in code
-        var registeredJobNames = _distributedBackgroundJobs.Select(x => x.Name).ToHashSet();
-        IEnumerable<DistributedBackgroundJobModel> jobsToRemove = existingJobs.Where(x => registeredJobNames.Contains(x.Name) is false);
+        // Batch insert new jobs
+        if (jobsToAdd.Count > 0)
+        {
+            _distributedJobRepository.Add(jobsToAdd);
+        }
+
+        // Batch delete jobs that are no longer registered
+        var jobsToRemove = existingJobs
+            .Where(x => !registeredJobsByName.ContainsKey(x.Name))
```

Comment:
Not a fan of the `!` notation, but feel free to ignore 🤣 
```suggestion
            .Where(x => registeredJobsByName.ContainsKey(x.Name) is false)
```